common/reconciliation/store/blobstoreWriter.go

```diff
@@ -106,9 +114,35 @@ func getBlobstoreWriteFn(
 			},
 		}
 
-		ctx, cancel := context.WithTimeout(context.Background(), Timeout)
-		defer cancel()
-		if _, err := client.Put(ctx, req); err != nil {
+		operation := func() error {
+			ctx, cancel := context.WithTimeout(context.Background(), Timeout)
+			defer cancel()
+			_, err := client.Put(ctx, req)
+			return err
+		}
+		// Using the ThrottleRetry struct and its Do method to implement the retry logic in the getBlobstoreWriteFn.
+		// This struct offers a way to retry operations with a specified policy and also to throttle retries if necessary.
+		retryPolicy := backoff.NewExponentialRetryPolicy(InitialRetryDelay)
+		retryPolicy.SetMaximumInterval(MaxRetryDelay)
+		retryPolicy.SetExpirationInterval(Timeout)
+		// Setting the attempts to 3 as a precaution. If we don't see any significant latency we can remove this config.
+		retryPolicy.SetMaximumAttempts(MaxRetries)
+
+		throttlePolicy := backoff.NewExponentialRetryPolicy(InitialRetryDelay)
+		throttlePolicy.SetMaximumInterval(MaxRetryDelay)
+		throttlePolicy.SetExpirationInterval(Timeout)
+
+		throttleRetry := backoff.NewThrottleRetry(
+			backoff.WithRetryPolicy(retryPolicy),
+			backoff.WithThrottlePolicy(throttlePolicy),
+			backoff.WithRetryableError(func(err error) bool {
+				return true // assuming all errors are retryable
+			}),
+		)
```

Comment:
general pattern elsewhere worth following: inject the policies.  they're not mutated, so you can reuse them.
this also lets you modify them for tests, to fake time and whatnot.

otherwise, while ThrottleRetry does seem like a better option, I'm not sure it's relevant here.  and if it is, it probably deserves a different policy: 
- it only changes behavior on `*types.ServiceBusyError`, which is probably not being returned by the blobstore.  or is it?
- the throttle-policy only advances when a throttle response is returned, using the same for both will mean it just always uses the non-throttle policy because it's always going to be the same delay or greater.

Reply:
I thought using throttling will be future proof in case we discover some error types that do not require retries so left it there. Should I remove it?

Reply:
non-retryable errors aren't throttle-retryable either, so in that sense: no.
throttle-retry in general: yeah, it's a bit more future-proof I think.  basically fine to leave it.

it's a bit more future-proof in that returning a ServiceBusy error is more likely to be handled reasonably / more easily adjusted / etc.  with things at the same value it doesn't make a difference, but it can at least be centrally adjusted, and e.g. one of the things we'd like to change is to allow busy-errors to define a minimum duration to wait before retrying -> that'd more likely get inherited here, which is good.